### DISCUSSION TITLE
Time Windows without disjunction

### DISCUSSION THREAD

**Discussion by mertbakir** [google/or-tools#2595] [Routing (and legacy CP) questions] [answered]
I wonder if we can set time windows but without disjunction. I want to apply a certain cost function if visit is not in the defined time window instead dropping the node.

Is there a way to achieve this?

I am working on this example: https://github.com/google/or-tools/blob/master/ortools/constraint_solver/samples/cvrptw.py

**Comment by Mizux** [comment, marked as answer]
Try to use this method from the `RoutingDimension` class:
https://github.com/google/or-tools/blob/b37d9c786b69128f3505f15beca09e89bf078a89/ortools/constraint_solver/routing.h#L2514-L2523

**Reply by mertbakir** [reply]
Thank you very much.
